src/discord-cluster-manager/report.py

```diff
@@ -77,40 +100,44 @@ async def _generate_crash_report(thread: discord.Thread, run: RunResult):
         message += (
             f"exited with error code **{run.exit_code}** after {float(run.duration):.2f} seconds."
         )
+    reporter.add_text(message)
 
     if len(run.stderr.strip()) > 0:
-        message = await _send_split_log(thread, message, "Program stderr", run.stderr.strip())
+        reporter.add_log("Program stderr", run.stderr.strip())
 
     if len(run.stdout.strip()) > 0:
-        message = await _send_split_log(thread, message, "Program stdout", run.stdout.strip())
+        reporter.add_log("Program stdout", run.stdout.strip())
```

Comment:
how come you don't need to await this anymore?

Reply:
we only need to await when we interact with discord; the new function just adds data to the reporter.
Things only get sent to discord once we reach the display_report function, which is still a coroutine